roxctl/central/debug/download_diagnostics_test.go

```diff
@@ -0,0 +1,105 @@
+package debug
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/roxctl/common/environment"
+	"github.com/stackrox/rox/roxctl/common/flags"
+	io2 "github.com/stackrox/rox/roxctl/common/io"
+	"github.com/stackrox/rox/roxctl/common/printer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDownloadDiagnosticsTimeoutReached(t *testing.T) {
+	shutdownServer := make(chan struct{}, 1)
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		<-shutdownServer
+	}))
+	defer server.Close()
+	defer close(shutdownServer)
+
+	_, stdErr, err := executeDiagnosticsCommand(t, server.URL, 10*time.Millisecond, "")
+
+	require.Error(t, err)
+	assert.True(t, isTimeoutError(err))
+	assert.Contains(t, stdErr.String(), "Timeout has been reached while creating diagnostic bundle")
+}
+
+func TestDownloadDiagnosticsServerError(t *testing.T) {
+	expectedErrorStr := "test-server-error"
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.WriteHeader(http.StatusInternalServerError)
+		_, _ = rw.Write([]byte(expectedErrorStr))
+	}))
+	defer server.Close()
+
+	_, stdErr, err := executeDiagnosticsCommand(t, server.URL, 20*time.Second, "")
```

Comment:
What will happen if we pass 0 or negative timeout? Is it possible?

Reply:
0 or negative timeout will result in http client having no timeout(I think it will still have connection timeout though)
I think that works that way because of how http client creates deadline:
https://github.com/golang/go/blob/master/src/net/http/client.go#L188-L193
 https://github.com/golang/go/blob/master/src/net/http/client.go#L345-L347